src/ol/interaction/selectinteraction.js

```diff
@@ -156,7 +211,9 @@ ol.interaction.Select.handleEvent = function(mapBrowserEvent) {
         features.item(0) == selected[0]) {
       // No change
     } else {
+      change = true;
       if (features.getLength() !== 0) {
+        Array.prototype.push.apply(deselected, features.getArray());
```

Comment:
To avoid hitting the maximum function argument length, you should do
```js
deselected = [].concat(features.getArray());
```
Maybe not so elegant because it creates a new array, but it is safer.